File: leetcode/medium/Q94_BinaryTreeInorderTraversal.py
from typing import List
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    answer = []
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        self.answer = []
        if root:
            self.find(root)
        return self.answer

    def find(self, node: TreeNode):
        if node.left is None and node.right is None:
            self.answer.append(node.val)
            return

        if node.left is not None:
            self.find(node.left)
        self.answer.append(node.val)
        if node.right is not None:
            self.find(node.right)

File: src/easy/Q94_BinaryTreeInorderTraversal.java
package easy;

import java.util.ArrayList;
import java.util.List;

public class Q94_BinaryTreeInorderTraversal {

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
        }
    }
    class Solution {
        List<Integer> answer = new ArrayList<>();
        public List<Integer> inorderTraversal(TreeNode root) {
            if(root == null) return answer;
            if(root.left != null){
                inorderTraversal(root.left);
            }
            answer.add(root.val);
            if(root.right != null){
                inorderTraversal(root.right);
            }
            return answer;
        }
    }
}
